Mixed Swift and Objective C static libraries require bridging header

* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report
The new Swift static libraries feature available with Xcode 9 and CocoaPods 1.4.0 can be used for targets that mix Objective C and Swift. 

However, for these to be built correctly (Swift must be able to 'see' ObjC code), a Bridging Header must be provided and I can't find any documented feature that enables this.

A workaround can be achieved by using: `s.xcconfig = { 'SWIFT_OBJC_BRIDGING_HEADER' => 'RxCocoa/RxCocoa.h' }` or setting it manually, but it is less than ideal and IMHO it should be promoted to a full podspec specification so the integration is easier.

## What did you do?
Try to use CocoaPods 1.4.0 beta 2 to build a mixed library

## What did you expect to happen?
For the library to compile correctly

## What happened instead?
The compilation failed because the ObjC code wasn't available for the Swift code.

## CocoaPods Environment

### Stack

```
   CocoaPods : 1.4.0.beta.2
        Ruby : ruby 2.3.3p222 (2016-11-21 revision 56859) [universal.x86_64-darwin17]
    RubyGems : 2.5.2
        Host : Mac OS X 10.13.1 (17B48)
       Xcode : 9.1 (9B55)
         Git : git version 2.13.6 (Apple Git-96)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories :  master - https://github.com/CocoaPods/Specs.git @ 484789662cfb979e75360a0803919f9939b2912b          
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
platform :ios, '9.0'
use_frameworks!

target 'RxSample' do
  pod 'RxCocoa', :path => '../RxCocoa.podspec'
  pod 'RxSwift', :path => '../RxSwift.podspec'

  target "RxSampleTests" do
    inherit! :search_paths
      pod 'RxBlocking', :path => '../RxBlocking.podspec'
      pod 'RxTest', :path => '../RxTest.podspec'
  end
end
```

## Project that demonstrates the issue

Please go [here](https://github.com/BlurredSoftware/RxSwift/tree/static-lib) at the `static-lib` branch, `cd` into RxCocoapodsExample and do a pod init

I prepared a small test project to reproduce this issue: https://github.com/kustra/NVMDummyTestPod
`pod lib lint` gives the following error:
```
 -> NVMDummy (2.0.0)
    - ERROR | [iOS] xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - ERROR | [iOS] xcodebuild:  /Users/kustra/Development/.../NVMDummyTestPod/NVMDummy/MySWClass.swift:14:14: error: use of unresolved identifier 'NVMMyOCClass'

[!] NVMDummy did not pass validation, due to 2 errors.
You can use the `--no-clean` option to inspect any issue.
```

If I remove [s.static_framework = true](https://github.com/kustra/NVMDummyTestPod/blob/9844e1d5d76c4ae8b1729f670801dd48665a3e52/NVMDummy.podspec#L15) from the podspec, it passes validation.

pod version 1.4.0.rc.1
Xcode version 9.2 (9C40b)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,10 +8,13 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Enhancements
 
-
 * Add `--exclude-pods` option to `pod update` to allow excluding specific pods from update  
   [Oleksandr Kruk](https://github.com/0mega)
   [#7334](https://github.com/CocoaPods/CocoaPods/issues/7334)
+
+* Add support for mixed Objective-C and Swift static frameworks  
+  [Paul Beusterien](https://github.com/paulb777)
+  [#7213](https://github.com/CocoaPods/CocoaPods/issues/7213)
 
 * Improve `pod install` performance for pods with exact file paths rather than glob patterns  
   [Muhammed Yavuz Nuzumlalı](https://github.com/manuyavuz)

--- a/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb
+++ b/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb
@@ -50,9 +50,6 @@ module Pod
                   create_info_plist_file(target.info_plist_path, native_target, target.version, target.platform)
                 end
                 create_build_phase_to_symlink_header_folders
-                if target.static_framework?
-                  create_build_phase_to_move_static_framework_archive
-                end
               elsif target.uses_swift?
                 add_swift_static_library_compatibility_header_phase
               end
@@ -481,28 +478,6 @@ module Pod
           base="$CONFIGURATION_BUILD_DIR/$WRAPPER_NAME"
           ln -fs "$base/${PUBLIC_HEADERS_FOLDER_PATH\#$WRAPPER_NAME/}" "$base/${PUBLIC_HEADERS_FOLDER_PATH\#\$CONTENTS_FOLDER_PATH/}"
           ln -fs "$base/${PRIVATE_HEADERS_FOLDER_PATH\#\$WRAPPER_NAME/}" "$base/${PRIVATE_HEADERS_FOLDER_PATH\#\$CONTENTS_FOLDER_PATH/}"
-            eos
-          end
-
-          # Creates a build phase to put the static framework in the appropriate framework location
-          # Since Xcode does not provide template support for static library frameworks, we've built a static library
-          # of the form lib{LibraryName}.a. We need to move that to the framework location -
-          # {LibraryName}.framework/{LibraryName}.
-          #
-          # @return [void]
-          #
-          def create_build_phase_to_move_static_framework_archive
-            build_phase = native_target.new_shell_script_build_phase('Setup Static Framework')
-            build_phase.shell_script = <<-eos.strip_heredoc
-          mkdir -p "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.framework/Modules"
-          # The fat library archive is at a file symbolic link when archiving, so use -L option
-          rsync -tL "${BUILT_PRODUCTS_DIR}/lib${PRODUCT_NAME}.a" "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.framework/${PRODUCT_NAME}"
-          rsync -t "${MODULEMAP_FILE}" "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.framework/Modules/module.modulemap"
-          # If there's a .swiftmodule, copy it into the framework's Modules folder
-          rsync -tr "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}".swiftmodule "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.framework/Modules/" 2>/dev/null || :
-          # If archiving, Headers copy is needed
-          rsync -tr "${TARGET_BUILD_DIR}/${PRODUCT_NAME}.framework/Headers" "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.framework/" 2>/dev/null || :
-          rsync -tr "${TARGET_BUILD_DIR}/${PRODUCT_NAME}.framework/PrivateHeaders" "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.framework/" 2>/dev/null || :
             eos
           end
 

--- a/lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb
+++ b/lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb
@@ -81,10 +81,8 @@ module Pod
             end
 
             if target.requires_frameworks?
-              framework_name = target.product_module_name
               if target.static_framework?
-                settings['PUBLIC_HEADERS_FOLDER_PATH'] = framework_name + '.framework' + '/Headers'
-                settings['PRIVATE_HEADERS_FOLDER_PATH'] = framework_name + '.framework' + '/PrivateHeaders'
+                settings['MACH_O_TYPE'] = 'staticlib'
               end
             else
               settings.merge!('OTHER_LDFLAGS' => '', 'OTHER_LIBTOOLFLAGS' => '')

--- a/lib/cocoapods/target.rb
+++ b/lib/cocoapods/target.rb
@@ -74,7 +74,7 @@ module Pod
     #         #requires_frameworks?.
     #
     def product_type
-      requires_frameworks? && !static_framework? ? :framework : :static_library
+      requires_frameworks? ? :framework : :static_library
     end
 
     # @return [String] A string suitable for debugging.

--- a/spec/unit/installer/xcode/pods_project_generator/pod_target_installer_spec.rb
+++ b/spec/unit/installer/xcode/pods_project_generator/pod_target_installer_spec.rb
@@ -818,20 +818,6 @@ module Pod
                 build_phase.should.not.be.nil
               end
 
-              it 'creates a build phase to set up a static library framework' do
-                @pod_target.stubs(:static_framework?).returns(true)
-
-                @installer.install!
-
-                target = @project.native_targets.first
-                build_phase = target.shell_script_build_phases.find do |bp|
-                  bp.name == 'Setup Static Framework'
-                end
-                build_phase.shell_script.should.include?('swiftmodule')
-                build_phase.shell_script.should.include?('PrivateHeaders')
-                build_phase.should.not.be.nil
-              end
-
               it 'verifies that headers in build phase for static libraries are all Project headers' do
                 @pod_target.stubs(:requires_frameworks?).returns(false)
 

--- a/spec/unit/installer/xcode/pods_project_generator/target_installer_spec.rb
+++ b/spec/unit/installer/xcode/pods_project_generator/target_installer_spec.rb
@@ -61,27 +61,15 @@ module Pod
             }
           end
 
-          it 'verify header path for a static library framework' do
+          it 'verify static framework is building a static library' do
             @pod_target.stubs(:requires_frameworks?).returns(true)
             @pod_target.stubs(:static_framework?).returns(true)
             @installer.send(:add_target)
-            @installer.send(:native_target).resolved_build_setting('PUBLIC_HEADERS_FOLDER_PATH').should == {
-              'Release' => 'BananaLib.framework/Headers',
-              'Debug' => 'BananaLib.framework/Headers',
-              'Test' => 'BananaLib.framework/Headers',
-              'AppStore' => 'BananaLib.framework/Headers',
-            }
-          end
-
-          it 'verify private header path for a static library framework' do
-            @pod_target.stubs(:requires_frameworks?).returns(true)
-            @pod_target.stubs(:static_framework?).returns(true)
-            @installer.send(:add_target)
-            @installer.send(:native_target).resolved_build_setting('PRIVATE_HEADERS_FOLDER_PATH').should == {
-              'Release' => 'BananaLib.framework/PrivateHeaders',
-              'Debug' => 'BananaLib.framework/PrivateHeaders',
-              'Test' => 'BananaLib.framework/PrivateHeaders',
-              'AppStore' => 'BananaLib.framework/PrivateHeaders',
+            @installer.send(:native_target).resolved_build_setting('MACH_O_TYPE').should == {
+              'Release' => 'staticlib',
+              'Debug' => 'staticlib',
+              'Test' => 'staticlib',
+              'AppStore' => 'staticlib',
             }
           end
         end

